File: backend/app/models/trip.rb
class Trip < ActiveRecord::Base
  belongs_to :user
  validates :user, presence: true
  validates_associated :user
  scope :destination, -> (destination) { where("lower(destination) like ?", "%#{destination.downcase}%").order(:destination)}
  scope :comment, -> (comment) { where("lower(comment) like ?", "%#{comment.downcase}%").order(:destination)}
  scope :starts_after, -> (date) { where("start_date >= ?", date).order(:start_date)}
  scope :ends_before, -> (date) { where("end_date <= ?", date).order(:end_date)}
end

File: frontend/app/mixins/validations/trip.js
import Ember from 'ember';
import EmberValidations from 'ember-validations';

export default Ember.Mixin.create(EmberValidations, {

  validations : {
    'model.destination': {
      presence: true
    },
    'model.start_date': {
      presence: true    },
    'model.end_date': {
      presence: true
    }
  }


});
